### PR TITLE
Add associations

### PR DESCRIPTION
**Pull request by BertrandConxy**
# oop-school-library: Add associations

Hello ✋ ,
In this branch, I have done the following:
- I have implemented a `has-many / belongs-to` relationship between `Classroom` and `Student` classes
- I have implemented the `many-to-many` relationship between `Book` and `Person` classes using `Rental` class as intermediate.

### CODE REVIEW

File: book.rb [hunk 1]
```diff
@@ -0,0 +1,10 @@
+class Book
+  attr_accessor :title, :author
+  attr_reader :rentals
+
+  def initialize(title, author)
+    @title = title
+    @author = author
+    @rentals = []
+  end
```
Comment (Deepakdanger) [inline book.rb]: - [x] I would recommend to add the `many-to-many` (also `has-many-through`) relationship between `Person` and `Book` using the intermediate class `Rental`.
```ruby
class Book
  ...
  ...
  def add_rental(person, date)
    # the book itself should be sent as a parameter to create the new rental achieving the 'has-many' association_. 
    Rental.new(date, self, person)
  end
end
```

File: book.rb [hunk 2]
```diff
@@ -0,0 +1,10 @@
+class Book
+  attr_accessor :title, :author
+  attr_reader :rentals
```
Comment (Deepakdanger) [inline book.rb]: - [x] I would recommend adding both Getters and Setters for `@rentals`

File: person.rb [hunk 3]
```diff
@@ -1,7 +1,7 @@
 require './decorator'
 
 class Person < Nameable
-  attr_reader :id
+  attr_reader :id, :rentals, :parent_permission
   attr_accessor :name, :age
```
Comment (Deepakdanger) [inline person.rb]: - [x] Same here as well. I would recommend adding both Getters and Setters for `@rentals`

File: person.rb [hunk 4]
```diff
@@ -10,6 +10,7 @@ def initialize(age, name = 'unknown', parent_permission: true)
     @name = name
     @age = age
     @parent_permission = parent_permission
+    @rentals = []
```
Comment (Deepakdanger) [inline person.rb:13]: - [x] I would recommend to add the `many-to-many` (also `has-many-through`) relationship between `Person` and `Book` using the intermediate class `Rental`.
```ruby
class Person
  ...
  ...
  def add_rental(book, date)
    # the book itself should be sent as a parameter to create the new rental achieving the 'has-many' association_. 
    Rental.new(date, book, self)
  end
end
```

**Review by Deepakdanger — CHANGES_REQUESTED** [review]
Hi @BertrandConxy, 

Good job so far! 👏
There are some issues that you still need to work on to go to the next project but you are almost there!

##  To Highlight :100: 
- Good PR :heavy_check_mark: 
- Well descriptive Readme :heavy_check_mark: 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: book.rb (hunk 1 above)
Comment (BertrandConxy) [reply book.rb]: I have implemented it in recent commit

File: book.rb (hunk 2 above)
Comment (BertrandConxy) [reply book.rb]: resolved

File: person.rb (hunk 3 above)
Comment (BertrandConxy) [reply person.rb]: resolved

File: person.rb (hunk 4 above)
Comment (BertrandConxy) [reply person.rb:13]: I have implemented it in recent committ.

**Review by aliabbani — APPROVED** [review]
Hi @BertrandConxy ✋, 

Your project is complete! Great work on making the changes requested by the previous reviewer There is nothing else to say other than... it's time to merge it :shipit:  
Congratulations! 🎉 



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by BertrandConxy** [comment]
Thank you @aliabbani

**Pull request merged**